code/obj/mining_cloud_storage.dm

```diff
@@ -0,0 +1,372 @@
+			var/obj/O = over_object
+			src.output_target = O.loc
+			boutput(usr, "<span class='notice'>You set [src] to output on top of [O]!</span>")
+
+		else if (istype(over_object,/turf) && !over_object:density)
+			src.output_target = over_object
+			boutput(usr, "<span class='notice'>You set [src] to output to [over_object]!</span>")
+
+		else
+			boutput(usr, "<span class='alert'>You can't use that as an output target.</span>")
+		return
+
+	MouseDrop_T(atom/movable/O as mob|obj, mob/user as mob)
+
+		if (!O || !user)
+			return
+
+		if(!isliving(user))
+			boutput(user, "<span class='alert'>Only living mobs are able to use the storage container's quick-load feature.</span>")
+			return
+
+		if (!isobj(O))
+			boutput(user, "<span class='alert'>You can't quick-load that.</span>")
+			return
+
+		if(!DIST_CHECK(O, user, 1))
+			boutput(user, "<span class='alert'>You are too far away!</span>")
+			return
+
+		if(!src.accept_loading(user,1))
+			boutput(user,"<span class='alert'>The storage container's quick-load system rejects you!</span>")
+			return
+
+		else if (istype(O, /obj/storage/crate/)  || istype(O, /obj/storage/cart/))
+			var/obj/storage/crate/C = O
+			if(istype(C) && (C.welded || C.locked))
+				boutput(user, "<span class='alert'>You cannot load from a crate that cannot open!</span>")
+				return
+			var/obj/storage/cart/Ct = O
+			if(istype(Ct) && (Ct.welded || Ct.locked))
+				boutput(user, "<span class='alert'>You cannot load from a cart that cannot open!</span>")
+				return
+
+			user.visible_message("<span class='notice'>[user] uses [src]'s automatic loader on [O]!</span>", "<span class='notice'>You use [src]'s automatic loader on [O].</span>")
+			var/amtload = 0
+			var/rejected = 0
+			for (var/obj/item/raw_material/M in O.contents)
+				if(M.material.name != M.initial_material_name)
+					rejected += M.amount
+					continue
+				amtload += M.amount
+				src.load_item(M)
+
+			if (amtload)
+				boutput(user, "<span class='notice'>[amtload] ores loaded from [O]!</span>")
+			if(rejected)
+				boutput(user, "<span class='alert'>[src] rejects [rejected] anomalous ore(s).</span>")
+			else boutput(user, "<span class='alert'>No ore loaded!</span>")
+
+		else if (isitem(O))
+			quickload(user,O)
+		else
+			..()
+
+		src.updateUsrDialog()
+
+	proc/quickload(var/mob/living/user,var/obj/item/O)
+		if (!user || !O)
+			return
+		if(istype(O,/obj/item/raw_material/))
+			var/obj/item/raw_material/R = O
+			if(R.material.name != R.initial_material_name)
+				boutput(user, "<span class='alert'>[src] rejects the anomalous ore.</span>")
+				return
+		user.visible_message("<span class='notice'>[user] begins quickly stuffing [O] into [src]!</span>")
+		var/staystill = user.loc
+		for(var/obj/item/raw_material/M in view(1,user))
+			if (!M)
+				continue
+			if (M.type != O.type)
+				continue
+			if(!isSameMaterial(O.material,M.material))
+				continue
+			if(M.material.name != M.initial_material_name)
```

Comment:
See comment about material quality adding prefixes to materials' names.

Reply:
quality doesn't add prefixes to material.name; it adds prefixes to the raw_material's name